ui/apps/platform/src/Containers/AccessControl/AuthProviders/AuthProvidersList.tsx

```diff
@@ -67,8 +67,10 @@ function AuthProvidersList({ entityId, authProviders }: AuthProvidersListProps):
                     </Tr>
                 </Thead>
                 <Tbody>
-                    {authProviders.map(({ id, name, type, defaultRole, groups = [] }) => {
+                    {authProviders.map((authProvider) => {
+                        const { id, name, type, defaultRole, traits, groups = [] } = authProvider;
                         const typeLabel = getAuthProviderTypeLabel(type, availableProviderTypes);
+                        const isImmutable = traits?.mutabilityMode !== 'ALLOW_MUTATE';
```

Comment:
@dhaus67 Here is my only question about the final version of this contribution:
* AuthProviderForm.tsx above consistently guards for the possibility that `traits` might have `undefined` value, for example:
     `values.traits && values.traits?.mutabilityMode !== 'ALLOW_MUTATE'`
* AuthProvidersList.tsx here seems double-minded because `?` implies the property might not exist, but `traits?.mutabilityMode !== 'ALLOW_MUTATE'` means `isImmutable` is true `undefined !== 'ALLOW_MUTATE'` in that case. But `traits` property absent means mutable, correct?
* authSagas.js similar `!(remaining.traits?.mutabilityMode !== 'ALLOW_MUTATE')` because `!(undefined !== 'ALLOW_MUTATE')` is `!true` for anonymous mutable condition, correct?
* AuthService seems similar to List above to return without saving if `traits` property is absent.

What do you think about consistent call of a helper function from AuthService.ts?

```ts
export function getIsAuthProviderImmutable(authProvider: AuthProvider) {
    return 'traits' in authProvider && authProvider.traits.mutabilityMode !== 'ALLOW_MUTATE';
}
```

Reply:
That's a good suggestion, I added the helper function + refactored the existing conditions for the mutability mode.